curve25519-dalek/src/field.rs

```diff
@@ -303,6 +309,51 @@ impl FieldElement {
     pub(crate) fn invsqrt(&self) -> (Choice, FieldElement) {
         FieldElement::sqrt_ratio_i(&FieldElement::ONE, self)
     }
+
+    #[cfg(feature = "digest")]
+    /// Hash_to_field as described in hash_to_curve standard
+    pub fn hash_to_field<D>(bytes: &[&[u8]], domain_sep: &[&[u8]]) -> Self
+    where
+        D: BlockSizeUser + Default + FixedOutput<OutputSize = U64> + HashMarker,
+    {
+        let l_i_b_str = 48u16.to_be_bytes();
+        let z_pad = vec![0u8; D::BlockSize::USIZE];
+
+        let mut hasher = D::new().chain_update(z_pad);
+
+        for slice in bytes {
+            hasher = hasher.chain_update(slice);
+        }
+
+        hasher = hasher.chain_update(l_i_b_str).chain_update([0u8]);
+
+        let mut domain_sep_len = 0usize;
+        for slice in domain_sep {
+            hasher = hasher.chain_update(slice);
+            domain_sep_len = domain_sep_len
+                .checked_add(slice.len())
+                .expect("Unexpected overflow from domain separator's size.");
+        }
```

Comment:
Checking for overflow on slice length with a `usize` is unnecessary, you would run out of memory before it overflows.